.github/workflows/ci.yml

```diff
@@ -712,3 +725,166 @@ jobs:
+              echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            LATEST_REMOTE_SHA=$(jq -r '.[0].sha' __response.json)
+            echo "LATEST_REMOTE_SHA is $LATEST_REMOTE_SHA"
+            echo "LATEST_LOCAL_SHA is $LATEST_LOCAL_SHA"
+            if [[ $LATEST_REMOTE_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because latest remote sha is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+            # Also validate the sha matches GITHUB_SHA, which is what gha-tag-release will use
+            if [[ $GITHUB_SHA != $LATEST_LOCAL_SHA ]]; then
+              echo "Not patch releasing because GITHUB_SHA is not equal to latest local sha"
+              DO_RELEASE=0
+            fi
+          fi
+
+          # Check is there is an existing tag on the branch using GitHub API
+          # Note cannot use local `git tag` because actions/checkout by default will not checkout tags
+          # and you need to checkout full history in order to get them
+          LATEST_TAG=""
+          NEXT_TAG=""
+          if [[ $DO_RELEASE == "1" ]]; then
+            # https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28
+            RESP_CODE=$(curl -w %{http_code} -s -o __response.json \
+            -X GET https://api.github.com/repos/${GITHUB_REPOSITORY}/tags?per_page=100 \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ github.token }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            )
+            if [[ $RESP_CODE != "200" ]]; then
+              echo "Unable to read list of commits - HTTP response code was $RESP_CODE"
+              exit 1
+            fi
+            # Get the latest tag
+            # Escape for regex e.g. 2.0 => 2\.0
+            ESCAPED_REF_NAME="${GITHUB_REF_NAME/./"\."}"
+            LATEST_TAG=$(jq -r '.[].name' __response.json | grep -E ^$ESCAPED_REF_NAME\.[0-9]+$ | sort -V -r | head -n 1)
+            echo "LATEST_TAG is $LATEST_TAG"
+            echo "latest_tag=$LATEST_TAG" >> $GITHUB_OUTPUT
+            if ! [[ $LATEST_TAG =~ ([0-9]+)\.([0-9]+)\.([0-9]+) ]]; then
+              echo "Not patch releasing because cannot find a matching semver tag on the branch"
+              DO_RELEASE=0
+            else
+              MAJOR=${BASH_REMATCH[1]}
+              MINOR=${BASH_REMATCH[2]}
+              PATCH=${BASH_REMATCH[3]}
+              NEXT_TAG="$MAJOR.$MINOR.$((PATCH+1))"
+              echo "NEXT_TAG is $NEXT_TAG"
+              echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
+            fi
+          fi
+
+          # Check if there is anything relevant to release using GitHub API
+          # The API results include all merged pull-requests, not commits
+          # Pull-requests prefixed with MNT or DOC will not be considered relevant for releasing
+          if [[ $DO_RELEASE == "1" ]]; then
```

Comment:
Given the exit above, we don't need this condition anymore.

Reply:
I'd prefer to keep the variable rather than switch to a function